ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFDateFormat.java

```diff
@@ -111,17 +123,18 @@ public Object evaluate(DeferredObject[] arguments) throws HiveException {
     // the function should support both short date and full timestamp format
     // time part of the timestamp should not be skipped
     Timestamp ts = getTimestampValue(arguments, 0, tsConverters);
+
     if (ts == null) {
       Date d = getDateValue(arguments, 0, dtInputTypes, dtConverters);
       if (d == null) {
         return null;
       }
       ts = Timestamp.ofEpochMilli(d.toEpochMilli(id), id);
     }
-
-
-    date.setTime(ts.toEpochMilli(id));
-    String res = formatter.format(date);
+    Timestamp ts2 = TimestampTZUtil.convertTimestampToZone(ts, timeZone, ZoneId.of("UTC"));
```

Comment:
No need to convert first convert timeZone to UTC and UTC to timeZone. Timestamp class internally use localDateTime which is timezone less and only depict the timestamp.

Reply:
Conversion is required. I have tested without conversion and we get wrong results

Reply:
Add comments on why this conversion is needed.

Reply:
done

Reply:
@guptanikhil007 @sankarh  We don't have to convert timeZone to UTC and UTC to timeZone. Since entire computation is done on localDateTime which is timezone less. So we just need to add the Time-zone to localDateTime.

We can do that as follows - 

formatter.format(ZonedDateTime.ofInstant(Instant.ofEpochSecond(ts.toEpochSecond(), ts.getNanos()), ZoneOffset.UTC)
        .withZoneSameLocal(timeZone))

Reply:
Not required